### PR TITLE
Rebase multiple-host-names onto the switched design

### PR DESCRIPTION
**Pull request by notriddle**

### CODE REVIEW

**Comment by jankeromnes** [comment]
Thanks a lot for rebasing https://github.com/JanitorTechnology/janitor/pull/188! :+1:

Please also make sure to address @bnjbvr's last night if you can.

**Comment by notriddle** [comment]
Fixed nits :+1:

File: app.js [hunk 1]
```diff
@@ -83,13 +83,18 @@ boot.executeInParallel([
     next();
   });
 
+  // Compute canonical resource URLs with a server middleware.
+  app.handle((request, response, next) => {
+    request.canonicalUrl = 'https://' + hostnames[0] + request.url;
```
Comment (jankeromnes) [inline app.js]: Self-nit: Don't we lose URL fragments by doing this? (E.g. canonical of `https://jntr.io/blog/#janitor-news-6` would become `https://janitor.technology/blog/` no?)

File: lib/routes.js [hunk 2]
```diff
@@ -96,6 +98,8 @@ exports.designPage = function (response, user = null) {
 const landingSection = camp.template([
   './templates/landing.html',
 ]);
+exports.landingPage = function (request, response) {
+  const { user, canonicalUrl } = request;
```
Comment (jankeromnes) [inline lib/routes.js]: Nit: These two lines look like an unintended copy/paste. Please remove them.

File: lib/routes.js [hunk 3]
```diff
@@ -153,19 +159,23 @@ exports.landingPage = function (response, user = null) {
...
   const title = 'API Reference';
   const htmlReference = api.toHTML();
 
   response.template({
     htmlReference,
     title,
     user,
+    canonicalUrl,
     scripts: [],
+
```
Comment (jankeromnes) [inline lib/routes.js]: Nit: Why?

File: lib/routes.js [hunk 4]
```diff
@@ -153,19 +159,23 @@ exports.landingPage = function (response, user = null) {
...
     stylesheets: [
       '/css/api-reference.css'
     ],
   }, [
+
```
Comment (jankeromnes) [inline lib/routes.js]: Nit: Why?

File: lib/routes.js [hunk 5]
```diff
@@ -229,16 +243,19 @@ exports.projectsPage = function (response, user = null) {
   ]);
 };
 
+
```
Comment (jankeromnes) [inline lib/routes.js]: Nit: Please remove this empty line.

File: lib/routes.js [hunk 6]
```diff
@@ -338,6 +362,7 @@ exports.settingsPageOld = async function (request, response, section, user) {
 
   const defaultConfigurations = Object.keys(configurations.defaults);
   const { username } = user.keys.github || {};
+
```
Comment (jankeromnes) [inline lib/routes.js]: Hmm, ok sure.

File: lib/routes.js [hunk 7]
```diff
@@ -503,6 +537,7 @@ exports.notFoundPage = function (response, user) {
   ]);
 };
 
+
```
Comment (jankeromnes) [inline lib/routes.js]: Nit: Please remove this empty line.

File: templates/header.html [hunk 8]
```diff
@@ -6,7 +6,8 @@
     <meta name="description" content="The fastest development system in the world.">
     <meta name="theme-color" content="#ffffff">
     <meta name="viewport" content="width=device-width, initial-scale=1">
-    <title>{{= title in html}}{{if title then {{ – }}}}Janitor</title>
+    <title> {{= title in html}}{{if title then {{ • }}}}Janitor </title>
```
Comment (jankeromnes) [inline templates/header.html]: Nit: Please consult with the person who switched from `•` to `–` before reverting it.

File: join.js [hunk 9]
```diff
@@ -15,15 +15,15 @@ const proxyHeuristics = require('./lib/proxy-heuristics');
...
-const hostname = db.get('hostname', 'localhost');
+// Add your actual server hostnames in './db.json':
+const hostnames = db.get('hostnames', [ 'localhost' ]);
 
-if (!hostname || hostname === 'localhost') {
-  throw new Error('Cannot join cluster as [hostname = ' + hostname + ']: ' +
-    'please fix the hostname in ./db.json and try again');
+if (!hostnames || hostnames[0] === 'localhost') {
+  throw new Error(`Cannot join cluster as [hostname = ${hostnames[0]}: ` +
+    `please fix the hostname in ./db.json and try again`);
```
Comment (jankeromnes) [inline join.js]: Nit: Maybe it would be clearer to specify "please fix the *first* hostname in ./db.json"?

**Comment by jankeromnes** [comment]
@notriddle Amazing, thanks!

Now there are just 3 linting problems left on Travis, please fix them as well:

> /home/travis/build/JanitorTechnology/janitor/app.js
>   124:13  error  'user' is assigned a value but never used  no-unused-vars
>   148:1   error  More than 1 blank line not allowed         no-multiple-empty-lines
>   401:1   error  More than 1 blank line not allowed         no-multiple-empty-lines
> ✖ 3 problems (3 errors, 0 warnings)
>   2 errors and 0 warnings potentially fixable with the `--fix` option.

**Comment by jankeromnes** [comment]
Also, could you please squash your fix commits into one? E.g. "Fix #188 nits; whitespace; merge mistake; unclear error message; linting problems" (or just "Fix nits", optionally with all commit messages listed in the commit description).

**Comment by jankeromnes** [comment]
(I took the liberty of fixing the linting problems, but feel free to squash that commit into your single fix commit.)

**Pull request merged**